Automattic/jetpack
# Adds a script to scan a repository for Core function calls.

This PR adds several new analyzer scripts that look for WordPress Core calls in a plugin code. 




#### Changes proposed in this Pull Request:

* Adds a new Analyzer script to find all function and method invocations in any plugin.
* Adds a new Analyzer script to find all WordPress Core function, method, static property and constant definition.
* Enables calculation of any plugin code dependency on WordPress Core functions using a combination of the previous two.

#### Jetpack product discussion


If you have a project that you need to quickly get a scope of possible changes or problematic code areas determined in, this tool can prove useful.

#### Does this pull request change what data or activity we track or use?


N/A

#### Testing instructions:





* Prepare a WordPress SVN checkout.
* Go to `packages/analyzer`
* Run `composer install`
* Run `composer run-script core-calls /path/to/a/plugin/folder /path/to/wordpress/src > output.csv`
* See the CSV dump of the function calls in the `output.csv` file.
* Find separate function call instances by filtering the csv: `cat output.csv | grep "wp_installing"`
* Filter out benevolent calls in the csv: `cat output.csv | grep -v "add_filter" | grep -v "__"` - any number of functions can be filtered out this way.

#### Proposed changelog entry for your changes:


* N/A
